### DISCUSSION TITLE
Access all nodes of certain type in AST

### DISCUSSION THREAD

**Discussion by mnaderhirn** [eclipse-langium/langium#1667] [Q&A] [answered]
What is the prefered way to get access to all nodes of a certain type in an AST

Is there an example for that?

**Comment by msujew** [comment, marked as answer]
Hey @mnaderhirn,

We have a `AstUtils` object exported from `langium` that exposes a few `stream...` methods. You should find one that fits your use case and filter the resulting stream using the generated `is<type>` functions.
